kdoctor/src/commonMain/kotlin/org/jetbrains/kotlin/doctor/diagnostics/GradleProjectDiagnostic.kt

```diff
@@ -0,0 +1,186 @@
+                "'${e.message}'"
+            )
+            return result.build()
+        }
+        val initScript = tempFile.substringBeforeLast("/") + "/init.gradle.kts"
+
+        system.execute("rm", initScript) //result code doesn't matter
+
+        val initScriptRenamed = system.execute("mv", tempFile, initScript).code
+        if (initScriptRenamed != 0) {
+            result.addFailure(
+                "Error $initScriptRenamed: impossible to prepare $initScript file",
+                "Check your file system write permissions"
+            )
+            return result.build()
+        }
+
+        Logger.d("Created $initScript")
+
+        val runGradle = system.execute(gradleBin, "-p", projectPath, "-I", initScript)
+
+        if (runGradle.code != 0) {
+            result.addFailure(
+                "Gradle error",
+                "Run in terminal '$gradleBin -p $projectPath --info'"
+            )
+            return result.build()
+        }
+
+        val gradleOutput = runGradle.output.orEmpty()
+        val pluginArtifacts = mutableListOf<PluginArtifact>()
+        val appliedPluginsMap = mutableMapOf<String, String>()
+        gradleOutput.lineSequence().forEach { l ->
+            if (l.contains(KDOCTOR_PREFIX)) {
+                val line = l.substringAfter(KDOCTOR_PREFIX)
+                if (line.startsWith("DEP ")) {
+                    //build script dependency
+                    val (group, name, version) = line.substringAfter("DEP ").split(":")
+                    pluginArtifacts.add(PluginArtifact(group, name, version))
+                } else {
+                    val (pluginId, pluginVersion) = line.split("=")
+                    //a plugin can be applied without a version
+                    if (!appliedPluginsMap.containsKey(pluginId) || pluginVersion != "null") {
+                        appliedPluginsMap[pluginId] = pluginVersion
+                    }
+                }
+            }
+        }
+
+        val appliedPlugins = findPluginVersions(appliedPluginsMap, pluginArtifacts).sortedBy { it.name }
+        if (appliedPlugins.isNotEmpty()) {
+            result.addSuccess(
+                "Gradle plugins:",
+                *appliedPlugins.map { it.name + ":" + it.version }.toTypedArray()
+            )
+        } else {
+            result.addWarning(
+                "Gradle plugins are not found"
+            )
+        }
+
+        result.addEnvironment(
+            EnvironmentPiece.Gradle(gradle.version),
+            *appliedPlugins.map { EnvironmentPiece.GradlePlugin(it.name, it.version) }.toTypedArray()
+        )
+
+        system.execute("rm", initScript) //result code doesn't matter
+
+        return result.build()
+    }
+
+    private fun printPluginsInitScript() = """
+        gradle.settingsEvaluated {
+            settings.pluginManagement.resolutionStrategy.eachPlugin {
+                println("$KDOCTOR_PREFIX${'$'}{requested.id}=${'$'}{requested.version}")
```

Comment:
nit: `$\{` a bit more readable for me than `${'$'}`

Reply:
It wouldn't work in this case. Because there should be `${...}` in the result text file.

Reply:
I meant `\${` actually. But you are right, it wouldn't work because there is no backslash escaping in raw strings 😞